Whisper.net/SpeechToTextClient/WhisperSpeechToTextClient.cs

```diff
@@ -0,0 +1,89 @@
+// Licensed under the MIT license: https://opensource.org/licenses/MIT
+
+using System.Runtime.CompilerServices;
+using System.Text;
+using Microsoft.Extensions.AI;
+
+#pragma warning disable MEAI001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+namespace Whisper.net;
+
+public sealed class WhisperSpeechToTextClient(string modelFileName) : ISpeechToTextClient
+{
+    private readonly WhisperFactory _factory = WhisperFactory.FromPath(modelFileName);
+    private WhisperProcessor? _processor;
+
+    public void Dispose()
+    {
+        _processor?.Dispose();
+        _factory?.Dispose();
+    }
+
+    public object? GetService(Type serviceType, object? serviceKey = null)
+    {
+        throw new NotImplementedException();
+    }
+
+    public async IAsyncEnumerable<SpeechToTextResponseUpdate> GetStreamingTextAsync(Stream audioSpeechStream, SpeechToTextOptions? options = null, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        if (audioSpeechStream is null)
+        {
+            throw new ArgumentNullException(nameof(audioSpeechStream));
+        }
+
+        this._processor ??= options.BuildWhisperProcessor(_factory);
```

Comment:
Is it possible that a client would be called with multiple options (e.g. initially for some language, and then for another)?

Also, the processor is not thread-safe. If it is cached, and multiple callers will call this method in parallel, the results might be interleaved.

I think it is better to create a processor for every call here and dispose it after the processing is done. We can keep and reuse the same `WhisperFactory` which will allow the reusal of the Model weights so they are not loaded multiple times in memory.